python/ray/autoscaler/_private/command_runner.py

```diff
@@ -701,7 +701,55 @@ def _docker_expand_user(self, string, any_char=False):
 
         return string
 
-    def run_init(self, *, as_head, file_mounts):
+    def _check_if_container_restart_is_needed(
+            self, image: str, cleaned_bind_mounts: Dict[str, str]) -> bool:
+        re_init_required = False
+        running_image = self.run(
+            check_docker_image(self.container_name),
+            with_output=True,
+            run_env="host").decode("utf-8").strip()
+        if running_image != image:
+            logger.error(f"A container with name {self.container_name} " +
+                         f"is running image {running_image} instead " +
+                         f"of {image} (which was provided in the YAML")
+        mounts = self.run(
+            check_bind_mounts_cmd(self.container_name),
+            with_output=True,
+            run_env="host").decode("utf-8").strip()
+        try:
+            active_mounts = json.loads(mounts)
+            active_remote_mounts = {
+                mnt["Destination"]
+                for mnt in active_mounts
+            }
+            # Ignore ray bootstrap files.
+            requested_remote_mounts = {
+                self._docker_expand_user(remote)
+                for remote in cleaned_bind_mounts.keys()
+            }
+            unfulfilled_mounts = (
+                requested_remote_mounts - active_remote_mounts)
+            if unfulfilled_mounts:
+                try:
+                    re_init_required = cli_logger.confirm(
+                        False, "This Docker Container is already Running, Do "
+                        "you want to restart the Docker container on "
+                        "this node?")
```

Comment:
when I ran this, it printed with some identation:
```
Shared connection to 52.27.65.236 closed.
    This Docker Container is already Running, Do you want to restart the Docker container on this node? Confirm [y/N]:
```
1) Can you remove the space before "This Docker Container..."?
2) For file mount in the product, we will reach this in the product because we are trying to upload mounts on a running docker. How can we interface with this from the CLI and from the code? (How can I avoid this prompt message by passing it in the command itself?)

Reply:
1. I'm not 100% sure how to remove that space (it is added from somewhere inside the clil_logger).

2. The current behavior is to default to restarting if running in a non-interactive context (i.e. the autoscaler). I can use the autoscaler's `-y` functionality if I add that argument to the `run_init` interface.

Reply:
Lets double check with @pcmoritz and @nikitavemuri to comment on whether this behavior would affect their work on handling file mounts.